Add optional documentation link to help text

akshaydewan opened the pull request:
Related to #6362 

Description: A lot of fields in the pipeline config page have a link to more documentation. This adds an optional property called `docLink` to the help text.

Example:
<img width="303" alt="Screenshot 2019-08-26 at 17 51 16" src="https://user-images.githubusercontent.com/6024038/63690391-223b3d00-c82a-11e9-84cb-c347c660eb11.png">
`More...` opens the linked gocd documentation in a new tab

GaneshSPatil commented:
@akshaydewan -- The "more" text initially gave me an impression that more information will be rendered inline. (similar to "show more.." and "show less..")

Can we change the wording to "Learn more"?

So:
- "more.."/"less.." - shows inline information.
- "learn more" - links to the external documentation.

Or is there a better way to reduce the confusion between inline help text vs external links?

GaneshSPatil commented:
\cc: @streisguth  @marques-work -- WDYT?

akshaydewan commented:
"Learn more..." seems ok to me.

ketan commented on server/webapp/WEB-INF/rails/webpack/views/components/forms/input_fields.tsx:
```diff
@@ -202,9 +205,16 @@ class HelpText extends MithrilViewComponent<HelpTextComponentAttrs> {
 
+  static maybeDocLink(docLink?: string) {
+    if (!_.isEmpty(docLink)) {
+      return <span>&nbsp;<Link href={docsUrl(docLink)} target="_blank">More&#8230;</Link></span>;
```
See https://www.owasp.org/index.php/Reverse_Tabnabbing. Perhaps make the `Link` component smart enough, based on the `href` attribute to add additional attrs?

akshaydewan replied:
Thanks for pointing that out. About the `Link` component, do you mean checking if the link specifies `target` and then adding `rel="noopener noreferrer"` if needed?

ketan replied:
What I meant is to make the `Link` component smart enough to specifically detect external links (`http(s)://`) and, based on that, automatically add the `target="_blank"` and `rel="noopener noreferrer"` attributes.

akshaydewan commented:
I've made some changes 
- Added the `rel="noopener noreferrer"` when a `target` is specified
- Changed the text to "Learn More"
- Used the external link icon instead of ellipsis
<img width="338" alt="Screenshot 2019-08-27 at 11 53 07" src="https://user-images.githubusercontent.com/6024038/63745999-41d17480-c8c1-11e9-9381-c47bb1f8428b.png">
